particles/Tutorial/Javascript/Demo/source/TicTacToeBoard.js

```diff
@@ -0,0 +1,110 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+'use strict';
+
+/* global defineParticle */
+
+defineParticle(({DomParticle, html}) => {   
+
+const template = html`
+  <style>
+    .grid-container {
+      display: grid;
+      grid-template-columns: 50px 50px 50px;
+      grid-column-gap: 0px;
+    }
+
+    .row {
+      content: "";
+      display: table;
+      clear: both;
+      display: inline-flex;
+    }
+
+    .butt {
+      border: 1px outset blue;
+      background-color: lightBlue;
+      height:50px;
+      width:50px;
+      cursor:pointer;
+    }
+
+    .butt:hover {
+      background-color: blue;
+      color:white;
+    }
+  </style>
+  <div class="grid-container">
+      <button class="butt" type="button" on-click="onClick" value=1>
```

Comment:
You can try playing with $template to reduce repetition, it would go something like this:
```
const template = html`
  <style>
  (...) 
  </style>
  <div class="grid-container">
    {{buttons}}
  </div>

  <template button>
    <button class="butt" type="button" on-click="onClick" value={{value}}> // You may need $={{value}} here, I'm not sure.
      <span>{{spot}}</span>
    </button>
  </template>
`

(...)
render(...) {
  return {
    buttons: {
      $template: 'button',
      models: arr.map((spot, index) => ({spot, value: index + 1}))
    }
  }
}
```

There are more examples in out `particle/` directory, just search for $template.

Reply:
Much nicer/shorter/cleaner. And it lets us have a template in the final demo, something it had been missing.